src/main/scala/swiss/dasch/domain/CsvService.scala

```diff
@@ -0,0 +1,119 @@
+    nrOfStillImageAssets,
+    nrOfMovingImageAssets,
+    nrOrOtherAssets,
+    prettyPrint(sizeOfStillImageOriginals),
+    prettyPrint(sizeOfStillImageDerivatives),
+    prettyPrint(sizeOfMovingImageOriginals),
+    prettyPrint(sizeOfMovingImageDerivatives),
+    prettyPrint(sizeOfMovingImageKeyframes),
+    prettyPrint(sizeOfOtherOriginals),
+    prettyPrint(sizeOfOtherDerivatives)
+  )
+}
+
+object AssetOverviewReportCsvRow {
+  val headerRow: List[String] = List(
+    "shortcode",
+    "totalNrOfAssets",
+    "nrOfStillImageAssets",
+    "nrOfMovingImageAssets",
+    "nrOfOtherAssets",
+    "sizeOfStillImageOriginals",
+    "sizeOfStillImageDerivatives",
+    "sizeOfMovingImageOriginals",
+    "sizeOfMovingImageDerivatives",
+    "sizeOfMovingImageKeyframes",
+    "sizeOfOtherOriginals",
+    "sizeOfOtherDerivatives"
+  )
+  def fromReport(rep: AssetOverviewReport): AssetOverviewReportCsvRow = {
+    def getFileSize(key: SupportedFileType, typ: String) =
+      rep.sizesPerType.sizes.get(key) match {
+        case None => FileSize(0)
+        case Some(other: SizeInBytesOther) =>
+          typ match {
+            case "orig"       => other.sizeOrig
+            case "derivative" => other.sizeDerivative
+            case _            => throw new IllegalArgumentException(s"Unknown type: $typ")
+          }
+        case Some(still: SizeInBytesMovingImages) =>
+          typ match {
+            case "orig"       => still.sizeOrig
+            case "derivative" => still.sizeDerivative
+            case "keyframes"  => still.sizeKeyframes
+            case _            => throw new IllegalArgumentException(s"Unknown type: $typ")
+          }
+      }
+
+    AssetOverviewReportCsvRow(
+      shortcode = rep.shortcode,
+      totalNrOfAssets = rep.totalNrOfAssets,
+      nrOfStillImageAssets = rep.nrOfAssetsPerType.getOrElse(StillImage, 0),
+      nrOfMovingImageAssets = rep.nrOfAssetsPerType.getOrElse(MovingImage, 0),
+      nrOrOtherAssets = rep.nrOfAssetsPerType.getOrElse(OtherFiles, 0),
+      getFileSize(StillImage, "orig"),
+      getFileSize(StillImage, "derivative"),
+      getFileSize(MovingImage, "orig"),
+      getFileSize(MovingImage, "derivative"),
+      getFileSize(MovingImage, "keyframes"),
+      getFileSize(OtherFiles, "orig"),
+      getFileSize(OtherFiles, "derivative")
+    )
+  }
+}
+
+final case class CsvService() {
+
+  private def createWriter(path: Path) = {
+    val acquire = ZIO.succeed(CSVWriter.open(path.toFile))
+    val release = (w: CSVWriter) => ZIO.succeed(w.close())
+    ZIO.acquireRelease(acquire)(release)
+  }
+
+  def writeReportToCsv(report: Seq[AssetOverviewReport], path: Path): Task[Path] =
+    ZIO.scoped {
+      for {
+        writer <- createWriter(path)
+        _      <- ZIO.succeed(writer.writeRow(AssetOverviewReportCsvRow.headerRow))
+        _ <- ZIO.foreachDiscard(report)(rep =>
+               ZIO.succeed(writer.writeRow(AssetOverviewReportCsvRow.fromReport(rep).toList))
```

Comment:
Weird formatting. Do we have `scalafmt` check here or should we have one in ingest too?

Reply:
https://github.com/dasch-swiss/dsp-ingest/blob/main/.github/workflows/ci.yml#L33